src/UartSerial.cpp

```diff
@@ -130,10 +136,43 @@ int UartSerial::read()
 
 void UartSerial::write(uint8_t byte_)
 {
-  ::write(_fd, &byte_, 1);
+  if(_fd != -1)
+    ::write(_fd, &byte_, 1);
 }
 
-void UartSerial::registerSerialEventCallback (serialEvent bytes_available_)
+void UartSerial::registerSerialEventCallback (serialEvent bytes_available_, void *context_)
 {
-  _bytes_available_callback = bytes_available_;
+  if(_fd == -1) return;
+
+  // For now only allow a single callback to be assigned
+  if(_bytes_available_callback == nullptr)
+  {
+    _bytes_available_callback = bytes_available_;
+    _callbackContext = context_;
+
+    // Setup poll struct
+    _fds[0].fd = _fd;
```

Comment:
total honesty dictates that I tell you I copied some code off the interwebs. Fixed now as a plain struct.